alf/algorithms/ppo_loss.py

```diff
@@ -99,26 +99,35 @@ def __init__(self,
         self._check_numerics = check_numerics
 
     def _pg_loss(self, training_info: TrainingInfo, advantages):
-        name_scope = 'PPOLoss'
-        importance_ratio, importance_ratio_clipped = value_ops.action_importance_ratio(
-            training_info, self._action_spec, name_scope,
-            self._importance_ratio_clipping, self._log_prob_clipping,
-            self._check_numerics, self._debug_summaries)
-        # Pessimistically choose the maximum objective value for clipped and
-        # unclipped importance ratios.
-        pg_objective = -importance_ratio * advantages
-        pg_objective_clipped = -importance_ratio_clipped * advantages
-        policy_gradient_loss = tf.maximum(pg_objective, pg_objective_clipped)
-
-        if self._debug_summaries and common.should_record_summaries():
-            with tf.name_scope(name_scope):
-                tf.summary.histogram('pg_objective', pg_objective)
-                tf.summary.histogram('pg_objective_clipped',
-                                     pg_objective_clipped)
-
-        if self._check_numerics:
-            policy_gradient_loss = tf.debugging.check_numerics(
-                policy_gradient_loss, 'policy_gradient_loss')
+        with tf.name_scope(self.__class__.__name__) as scope:
+            importance_ratio, importance_ratio_clipped = value_ops.action_importance_ratio(
```

Comment:
you don't need to call this under with .... as scope, you are passing scope as an argument. I better way is:
scope = tf.name_scope(...) as scope